testinfra/modules/addr.py

```diff
@@ -32,7 +32,8 @@ def is_reachable(self):
                 self._addr.name, self._port).rc == 0
 
         return self._addr.run(
-            "nc -w 1 -z %s %s", self._addr.name, self._port).rc == 0
+            "echo /dev/zero | nc -w 1 %s %s",
```

Comment:
Can you explain how this works ? Actually this seems to send a string "/dev/zero" to the socket which is not equivalent to the `-z` option.

```
$ nc -w 1 -z 127.0.0.1 22
(nothing is send to sshd)
$ echo /dev/zero | nc -w 1 127.0.0.1 22  
SSH-2.0-OpenSSH_7.9p1 Debian-10
Protocol mismatch.
```

Reply:
You right. I see 2 possible solutions.

1. Never use **nc** and always use **bash**
2. Check if package **netcat** is present whereas **nc** binary

Any preferences ?

Reply:
up @philpep  :)